tests/test_onedrive.py

```diff
@@ -532,7 +535,13 @@ def direct_api_raises_errors(action, path: str):
     with patch.object(OneDriveProvider, "_base_url", api.uri()):
         with patch.object(odp, '_direct_api', side_effect=direct_api_raises_errors):
             odp.reconnect()
-            log.info("namespaces: %s", odp.list_ns())
+            # personal namespace is the failsafe
+            namespaces = odp.list_ns()
+            assert namespaces[0].name == "Personal"
+            # namespace errors are saved and can be queried
+            errors = odp.list_ns(parent=Namespace("errors", "errors"))
```

Comment:
This seems to introspect into the error list unnecessarily. Create a constant in the class or in the module, and use the constant as the `parent= ...` value instead, so that you can change the implementation if you need. For example, if OneDrive decides to create their own "errors" namespace, you can change the value of the magic cookie to something else. Clients will need to update to the latest version of this library to fix the problem, but their code won't need to change

Reply:
Technically, parent doesn't even need to be an object of type Namespace... One possibility would be to derive a new `class ErrorNamespace(Namespace): pass` and instead of checking if the `parent.id == "errors"` check if `isinstance(parent, ErrorNamespace)` and then it shouldn't risk a conflict with the namespace ids.

Reply:
makes sense, it is a bit less of an easter egg that way.